ignite/pkg/cosmosgen/generate.go

```diff
@@ -104,23 +116,133 @@ func (g *generator) setup() (err error) {
 			if err != nil {
 				return err
 			}
+			includes := []string{}
+			if len(modules) > 0 {
 
+				includes, err = g.resolveIncludes(path) // For versioning issues, we do dependency/includes resolution per module
+				if err != nil {
+					return err
+				}
+			}
 			modulesInPath = ModulesInPath{
-				Path:    path,
-				Modules: modules,
+				Path: path, // Each go module
+				ModuleIncludes: ModuleIncludes{ // Has a set of sdk modules and a set of includes to build them
+					Includes: includes,
+					Modules:  modules,
+				},
 			}
 
 			if err := moduleCache.Put(cacheKey, modulesInPath); err != nil {
 				return err
 			}
 		}
 
-		g.thirdModules[modulesInPath.Path] = append(g.thirdModules[modulesInPath.Path], modulesInPath.Modules...)
+		g.thirdModules[modulesInPath.Path] = modulesInPath.ModuleIncludes
 	}
 
 	return nil
 }
 
+func (g *generator) getProtoIncludeFolders(modPath string) []string {
+	includePaths := []string{filepath.Join(modPath, g.protoDir)} // Add default protoDir and default includeDirs
+	for _, dir := range g.o.includeDirs {
+		includePaths = append(includePaths, filepath.Join(modPath, dir))
+	}
+	return includePaths
+}
+
+/*
+Leave this here for reference to improve resolution algorithm
+
+	func (g *generator) checkForProto(modpath string) (bool, error) {
+		err := filepath.WalkDir(modpath,
+			func(path string, _ fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				if filepath.Ext(path) == ".proto" {
+					return errFileFound
+				}
+				return nil
+			})
+		if err == errFileFound {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		return false, errFileNotFound
+	}
+*/
+func (g *generator) checkForBuf(modpath string) (string, error) {
+	var bufPath string = ""
```

Comment:
You can safely remove the empty value assign because it's the default one for the `string` type:

```suggestion
	var bufPath string
```